core/unit_test/TestMDTeamParallelism.hpp

```diff
@@ -0,0 +1,1660 @@
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_8D_MDThreadVectorRange<Right, Right>(dims);
+  }
+
+  {
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_3D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_3D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_3D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_3D_MDTeamVectorRange<Right, Right>(dims);
+
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_4D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_4D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_4D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_4D_MDTeamVectorRange<Right, Right>(dims);
+
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_5D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_5D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_5D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_5D_MDTeamVectorRange<Right, Right>(dims);
+
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_6D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_6D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_6D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_6D_MDTeamVectorRange<Right, Right>(dims);
+
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_7D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_7D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_7D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_7D_MDTeamVectorRange<Right, Right>(dims);
+
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_8D_MDTeamVectorRange<Left, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_8D_MDTeamVectorRange<Left, Right>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_8D_MDTeamVectorRange<Right, Left>(dims);
+    TestMDTeamParallelFor<TEST_EXECSPACE>::
+        test_parallel_for_8D_MDTeamVectorRange<Right, Right>(dims);
+  }
+
+  {
+    TestMDTeamParallelFor<TEST_EXECSPACE>::test_parallel_single_direction_test<
+        Left>(dims);
```

Comment:
No `test_parallel_single_direction_test<Right>(dims)`?

Reply:
Because the default direction is `Right`, it seems that we took it as granted and didn't add an explicit test for it.
But it makes sense to include an explicit test for that condition as well, so I added in a right-directional test in the latest commit as you suggested.